### PR TITLE
refactor(camera): alignment and docs

### PR DESCRIPTION
**Pull request by ferdicus**
* remove unnecessary/ obsolete props/ methods
* update docs

### CODE REVIEW

File: docs/Camera.md [hunk 1]
```diff
@@ -40,10 +42,8 @@
 | followZoomLevel | `number` | `none` | `false` | The zoomLevel on map while followUserLocation is set to `true` |
 | followPitch | `number` | `none` | `false` | The pitch on map while followUserLocation is set to `true` |
 | followHeading | `number` | `none` | `false` | The heading on map while followUserLocation is set to `true` |
-| triggerKey | `any` | `none` | `false` | FIX ME NO DESCRIPTION |
-| alignment | `array` | `none` | `false` | FIX ME NO DESCRIPTION |
+| triggerKey | `any` | `none` | `false` | Manually update the camera - helpful for when props did not update, however you still want the camera to move |
```
Comment (ClayC-WeatherBug) [inline docs/Camera.md:45]: I'm pretty sure I ran into something like this recently. Great to know it is an option 👍

File: javascript/components/Camera.js [hunk 2]
```diff
@@ -58,6 +58,11 @@ const SettingsPropTypes = {
      * Bottom camera padding for bounds
      */
     paddingBottom: PropTypes.number,
+
+    /**
+     * Callback that is triggered on user tracking mode changes
+     */
+    onUserTrackingModeChange: PropTypes.func,
```
Comment (ClayC-WeatherBug) [inline javascript/components/Camera.js:65]: Do we have typescript definitions for Camera that might also need updates, or does it still need to be updated?

**Review by ClayC-WeatherBug — APPROVED** [review]

File: javascript/components/Camera.js (hunk 2 above)
Comment (ferdicus) [reply javascript/components/Camera.js:65]: already typed :)

https://github.com/react-native-mapbox-gl/maps/blob/abafc67c8d3fbf550dc5820c85ce1973fc8a089f/index.d.ts#L445-L453

Comment (ClayC-WeatherBug) [reply javascript/components/Camera.js:65]: Ah - do we plan on converting the component files over to TS instead of doing just the index.d.ts?

Comment (ferdicus) [reply javascript/components/Camera.js:65]: yep, that's the plan:
* convert all js/ jsx => ts/ tsx
* add types/ interfaces
* run a tsc build and generate a `/lib` folder plus `index.d.ts`

**Review by mfazekas — APPROVED** [review]
I assume nothing in backward incompatible here, only non used code was removed, right?!

**Comment by ferdicus** [comment]
> I assume nothing in backward incompatible here, only non used code was removed, right?!

correct 👍

**Pull request merged**